### ISSUE TITLE
US banks data-importer

### ISSUE THREAD

**Issue by a7maadf** [firefly-iii/firefly-iii#7790]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)
- [X] My request is not listed as [a very good idea, but unfortunately...](https://docs.firefly-iii.org/firefly-iii/more-information/what-its-not/)
- [X] I've used [the search](https://github.com/firefly-iii/firefly-iii/issues?q=is%3Aissue) and this has not been requested before.

### Description

Giving that Salt Edge has dropped support of US banks, I was wondering what are the chances of using an alternative service like www.basiq.io ? They have a very cheap pricing ($0.5/month) and support over 140 bank institutions  

### Solution

read description

**Comment by JC5** [comment]
Looks great!

I can put this on the list to develop from the data importer, but I must admit I will not a lot of time to implement it. If you are a bit of a programmer though, it should not be hard to develop this for yourself 😉

**Comment by a7maadf** [comment]
I honestly don't have the best experience with PHP but I could certainly try you know ... come up with something. 

Thank you so much for this perfect piece of software. Will keep you updated

**Comment by dvankley** [comment]
@a7maadf If you're ok with using Plaid (even just a developer account), [this connector](https://github.com/dvankley/firefly-plaid-connector-2) might do what you want.

**Comment by Kage1** [comment]
> @a7maadf If you're ok with using Plaid (even just a developer account), [this connector](https://github.com/dvankley/firefly-plaid-connector-2) might do what you want.

I have tried Plaid and I must say it's very clunky. When I had it working the biggest issue I kept having was the connection would randomly break and I didn't know for a week or so. I had to go through trying to force the update and hopefully not end up with duplicates.

I just updated the Plaid quickstart and several bank connectors are showing Something Went Wrong please try again. So yea, Plaid is an option, but be prepared to tinker with it.

**Comment by dvankley** [comment]
@Kage1 was your experience using `firefly-plaid-connector-2`? Agreed that Plaid is fairly flaky, but some of those issues might have been shortcomings in the connector that I could try to shore up (better communication of broken accounts, etc.). If so feel free to file an issue or two on that project.

**Comment by Kage1** [comment]
It was with the original connector, not the newer #2.

What I just discovered while digging into the new connector yesterday is that the Plaid developer account can't be used for a lot of the institutions now. Only a approved paid prod account will work.

I haven't gotten far enough yet to say how well connector2 works.

**Comment by Kage1** [comment]
@dvankley 
The last time was with the original connector not #2.

The main issue now is that several banks will not allow connections from Dev accounts and only from prod. They throw an error message and the logs then docs indicate a full approved paid prod account is needed.

So now I'm deciding how far I want to dig into it.

**Comment by dahifi** [comment]
First day looking at Firefly, coming from GnuCash, anyways, just wasted time signing up for GoCardless but no good bc no US banks. So I guess I'll be taking a look at this plugin as well.

**Comment by curiousercreative** [comment]
> First day looking at Firefly, coming from GnuCash, anyways, just wasted time signing up for GoCardless but no good bc no US banks. So I guess I'll be taking a look at this plugin as well.

@dahifi does GNUCash have a US bank importer?

**Comment by Kage1** [comment]
Sadly no, the us banking system is a mess with nothing that is standardized for automation. There are several 3rd party companies that do not interoperate with each other. The services they do offer generally do not have a free tier.

**Comment by dvankley** [comment]
@Kage1 it's not a great option, but you could consider a Plaid paid account. They don't publish hard numbers, but what I've seen from others is $.30 per month, per Item for transaction data.

**Comment by curiousercreative** [comment]
Yeah, there is Plaid and SimpleFIN that I've seen importers with, neither is super straughtforward

**Comment by sbates130272** [comment]
When I looked at [www.basiq.io](http://www.basiq.io/) today it looks like only Australian and New Zealand banks are supported. Is this true or am I missnig something?

I am looking into options for importer support for Canadian banks. Like the US it is a bit of a mess right now.

**Comment by Kage1** [comment]
What I have ended up doing is setting a calendar reminder and sat morning I spend a few mins d/ling and importing. With the import profiles setup it works ok but I end up with a weeks lag in data. :/

**Comment by Snuupy** [comment]
https://www.reddit.com/r/mintuit/comments/17iq9gx/plaid_api_production_version_for_personal_use/karu5aj/

**Comment by dahifi** [comment]
I did some exploration on the Plaid side, even getting a development account with access to live data requires several hoops. 

Right now I'm planning on doing CSV exports, hopefully I can figure out some automations to make it less tedious.

**Comment by fear025** [comment]
For those investigating Plaid connectivity, please be aware that the free development environment is going away as of July 2024.
That means you'll be forced to use their production environment with their associated fees. Currently the fees are something like 30 cents USD per month per account.

Here's their announcement text:
Changes to our Development environment

Plaid will stop supporting its Development test environment by July 2024. If you have unused items in the Development environment, you’ll be provided with some credits to test in our Production environment in a limited capacity.

No action is required of you at this time.

What’s changing in 2024?
We’re working to provide developers and businesses with a simpler and more intuitive testing experience with Plaid’s products. As part of our work, we’ll be deprecating the Development test environment and introducing limited, free testing in our Production environment.

You can continue to test in the Development environment until we remove all existing Development items on June 20, 2024.

Please share this information with your technical team, and stay tuned for more updates when we’re ready for you to begin testing in Production.

**Comment by dvankley** [comment]
@fear025 That's good information, if disappointing. Where did you see that?

For the time being I'll probably just pay the couple bucks a month for the paid version. If anyone has experience with compelling alternatives, I'm open to possibly making another connector.

**Comment by fear025** [comment]
> @fear025 That's good information, if disappointing. Where did you see that?
> 
> For the time being I'll probably just pay the couple bucks a month for the paid version. If anyone has experience with compelling alternatives, I'm open to possibly making another connector.

@dvankley First, thanks so much for your work on the firefly-plaid-connector-2, it's been working great for me!

I copied and pasted that text from an email Plaid sent me on December 20th, 2023. 
The information can be confirmed on this plaid webpage: [https://plaid.com/docs/sandbox/](url)

**Comment by curiousercreative** [comment]
SimpleFIN Bridge is worth investigating, it's a layer on top of MX, Plaid's major competitor in the US.

**Comment by dahifi** [comment]
Thanks @curiousercreative I'm going to add this to my calendar next week: https://beta-bridge.simplefin.org/info/developers

**Comment by curiousercreative** [comment]
@dahifi there's an open issue on thiss repo [here](https://github.com/firefly-iii/firefly-iii/issues/5396) and there's repo for a similar feature for Actual Budget [here](https://github.com/lancepick/actual-simplefin-sync)

**Comment by dahifi** [comment]
Re. Plaid: So was looking through the [discussions](https://github.com/orgs/firefly-iii/discussions/7065) and came across this lovely contribution from @cskujawa that might be worth a look: https://github.com/cskujawa/youteller

**Comment by ConorIA** [comment]
I noticed another Canadian finance app is using Flinks, which also seems to support US banks and may be an alternative to Plaid worth exploring: https://flinks.com/about/

**Comment by Responseless** [comment]
Any update on how the basiq.io import is coming along @JC5 ?

**Comment by ConorIA** [comment]
An additional alternative that is super smooth with Actual Budget is SimpleFIN. It is a piece of cake to accesa the data from the app side too. https://www.simplefin.org/protocol.html

**Comment by JC5** [comment]
SimpleFIN is also integrated into the Firefly III data importer :).

**Comment by Responseless** [comment]
I'm an Aussie so none of these services work with our banks. Good news is the Australian Government is making all the instutions create public API's but that wont be available until 2026-27. See https://consumerdatastandardsaustralia.github.io/standards/ . Basiq.io is the one of the only ones I've found which supports Australian banks extensively.

**Comment by Kage1** [comment]
> SimpleFIN is also integrated into the Firefly III data importer :).

I saw it show up in the last Converter version, but thought it was mostly European banks only. To my surprise it supports all of the US banks that I use. I'm working through to set it up now.
